### PR TITLE
fix: The second param of the  allPaths/allKeys is redundant

### PR DESCRIPTION
**Pull request by helios741**
`prevIndex === -1 ? 0 : prevIndex + 1` and `prevIndex + 1` are completely equivalent, using the latter can also reduce the code

### CODE REVIEW

File: modules/createBrowserHistory.js [hunk 1]
```diff
@@ -183,10 +183,7 @@ function createBrowserHistory(props = {}) {
             window.location.href = href;
           } else {
             const prevIndex = allKeys.indexOf(history.location.key);
-            const nextKeys = allKeys.slice(
-              0,
-              prevIndex === -1 ? 0 : prevIndex + 1
-            );
+            const nextKeys = allKeys.slice( 0,prevIndex + 1 );
```
Comment (guidobouman) [inline modules/createBrowserHistory.js]: `(0, prevIndex + 1);`

File: modules/createHashHistory.js [hunk 2]
```diff
@@ -210,10 +210,7 @@ function createHashHistory(props = {}) {
           pushHashPath(encodedPath);
 
           const prevIndex = allPaths.lastIndexOf(createPath(history.location));
-          const nextPaths = allPaths.slice(
-            0,
-            prevIndex === -1 ? 0 : prevIndex + 1
-          );
+          const nextPaths = allPaths.slice( 0,  prevIndex + 1 );
```
Comment (guidobouman) [inline modules/createHashHistory.js]: See previous comment.

Comment (helios741) [reply modules/createHashHistory.js]: If there is a new version of history, isn't this PR unnecessary?

**Comment by pshrmn** [comment]
Closing in favor of #693.

**Pull request closed without merging**